[Bug] Rast_is_c_null_value causes -Werror=unused-value in C++

## Describe the bug

Rast_is_c_null_value and the related macros cause `error: statement has no effect [-Werror=unused-value]` with GCC 13 and C++17.

## To reproduce

I get this while compiling _r.viewshed_. Not for anything else.

## Screenshots

```
g++  -O0 -Wall -Wextra -Wpedantic -Werror -std=c++17 -ggdb  -I/home/vpetras/Projects/grass/code/grass/dist.x86_64-pc-linux-gnu/include -I/home/vpetras/Projects/grass/code/grass/dist.x86_64-pc-linux-gnu/include   -DUSER=\"vpetras\" -Wno-sign-compare -DPACKAGE=\""grassmods"\"   -I/home/vpetras/Projects/grass/code/grass/dist.x86_64-pc-linux-gnu/include -I/home/vpetras/Projects/grass/code/grass/dist.x86_64-pc-linux-gnu/include -DRELDIR=\"raster/r.viewshed\" -o OBJ.x86_64-pc-linux-gnu/grass.o -c grass.cpp
In file included from /home/vpetras/Projects/grass/code/grass/dist.x86_64-pc-linux-gnu/include/grass/raster.h:239,
                 from grass.h:43,
                 from grass.cpp:46:
grass.cpp: In function ‘void save_io_vis_and_elev_to_GRASS(IOVisibilityGrid*, char*, char*, float)’:
/home/vpetras/Projects/grass/code/grass/dist.x86_64-pc-linux-gnu/include/grass/defs/raster.h:404:31: error: statement has no effect [-Werror=unused-value]
  404 |     (*(const CELL *)(cellVal) == (CELL)0x80000000)
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
grass.cpp:977:17: note: in expansion of macro ‘Rast_is_c_null_value’
  977 |                 Rast_is_c_null_value(&((CELL *)elevrast)[j]);
      |                 ^~~~~~~~~~~~~~~~~~~~
/home/vpetras/Projects/grass/code/grass/dist.x86_64-pc-linux-gnu/include/grass/defs/raster.h:406:33: error: statement has no effect [-Werror=unused-value]
  406 |     (*(const FCELL *)(fcellVal) != *(const FCELL *)(fcellVal))
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
grass.cpp:981:17: note: in expansion of macro ‘Rast_is_f_null_value’
  981 |                 Rast_is_f_null_value(&((FCELL *)elevrast)[j]);
      |                 ^~~~~~~~~~~~~~~~~~~~
/home/vpetras/Projects/grass/code/grass/dist.x86_64-pc-linux-gnu/include/grass/defs/raster.h:408:33: error: statement has no effect [-Werror=unused-value]
  408 |     (*(const DCELL *)(dcellVal) != *(const DCELL *)(dcellVal))
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
grass.cpp:985:17: note: in expansion of macro ‘Rast_is_d_null_value’
  985 |                 Rast_is_d_null_value(&((DCELL *)elevrast)[j]);
      |                 ^~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make: *** [../..//include/Make/Compile.make:35: OBJ.x86_64-pc-linux-gnu/grass.o] Error 1
```

## System description

- Ubuntu 24.04.1 LTS (noble)
- gcc (Ubuntu 13.2.0-23ubuntu4) 13.2.0
- `-std=c++17`
- main branch

--- a/raster/r.viewshed/grass.cpp
+++ b/raster/r.viewshed/grass.cpp
@@ -450,6 +450,7 @@ AMI_STREAM<AEvent> *init_event_list(char *rastName, Viewpoint *vp,
     Rast_set_null_value(inrast[2], ncols, data_type);
 
     /*scan through the raster data */
+    // int isnull = 0;
     dimensionType i, j;
     double ax, ay;
     AEvent e;
@@ -483,7 +484,7 @@ AMI_STREAM<AEvent> *init_event_list(char *rastName, Viewpoint *vp,
             e.col = j;
 
             /*read the elevation value into the event */
-            Rast_is_null_value(&(inrast[1][j]), data_type);
+            // isnull = Rast_is_null_value(&(inrast[1][j]), data_type);
             e.elev[1] = inrast[1][j];
 
             /* adjust for curvature */
@@ -971,18 +972,19 @@ void save_io_vis_and_elev_to_GRASS(IOVisibilityGrid *visgrid, char *elevfname,
         for (j = 0; j < Rast_window_cols(); j++) {
 
             /* read the current elevation value */
+            // int isNull = 0;
 
             switch (elev_data_type) {
             case CELL_TYPE:
-                Rast_is_c_null_value(&((CELL *)elevrast)[j]);
+                // isNull = Rast_is_c_null_value(&((CELL *)elevrast)[j]);
                 elev = (double)(((CELL *)elevrast)[j]);
                 break;
             case FCELL_TYPE:
-                Rast_is_f_null_value(&((FCELL *)elevrast)[j]);
+                // isNull = Rast_is_f_null_value(&((FCELL *)elevrast)[j]);
                 elev = (double)(((FCELL *)elevrast)[j]);
                 break;
             case DCELL_TYPE:
-                Rast_is_d_null_value(&((DCELL *)elevrast)[j]);
+                // isNull = Rast_is_d_null_value(&((DCELL *)elevrast)[j]);
                 elev = (double)(((DCELL *)elevrast)[j]);
                 break;
             }

